app/register/controller.ts

```diff
@@ -73,7 +73,7 @@ export default class Register extends Controller.extend(registerQueryParams.Mixi
 
     setup({ queryParams }: { queryParams: RegisterQueryParams }) {
         if (queryParams.campaign) {
-            const matches = queryParams.campaign.match(/^(.*)-(.*)$/);
+            const matches = queryParams.campaign.match(/^(.*)-(.*)|(.*)$/);
```

Comment:
Hm, this regex is now equivalent to `/(.*)/`, which matches anything -- is that what we want? If so, no need for the regex at all.

Reading downward, `matches.length > 2` will be false for `campaign=prereg`, so all setup will be skipped -- i'm not sure this change changes anything?

Reply:
For `campaign=prereg`, `/^(.*)-(.*)|(.*)$/` returns `['prereg', undefined, undefined, 'prereg']` so `matches.length > 2` will actually be true. This condition is kind of pointless because `matches.length` will always be the number of capture groups + 1 for the whole string matched.

Reply:
(and yes, I was the one who added this condition)

Reply:
ohh, I forgot how `match` works with groups 👍

Reply:
Me too, apparently! 😆